authentik/lib/expression/evaluator.py

```diff
@@ -221,13 +251,8 @@ def handle_error(self, exc: Exception, expression_source: str):  # pragma: no co
 
     def validate(self, expression: str) -> bool:
         """Validate expression's syntax, raise ValidationError if Syntax is invalid"""
-        param_keys = self._context.keys()
         try:
-            compile(
-                self.wrap_expression(expression, param_keys),
-                self._filename,
-                "exec",
-            )
+            self.compile(expression)
```

Comment:
It's a bit of a shame that this won't validate whether the expression will run. For instance, it won't check that `import module` will work, because that relies on the builtin `__import__` being there, and we purposefully remove it when we call `exec`.